src/PlaywrightSharp/BrowserTypeBase.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace PlaywrightSharp
+{
+    /// <summary>
+    /// Base type for implementing <see cref="IBrowserType"/>.
+    /// </summary>
+    public abstract class BrowserTypeBase : IBrowserType
```

Comment:
With so many abstracts here... how about removing `IBrowserType` here, and moving `IBrowserType` to each class?
This class would be way more cleaner and we wouldn't have so many overrides on the final classes.

Reply:
that's what I did at first but then I noticed that `BrowserTypeBase` does implement some of the members on the interface.
maybe we could rename this to `BrowserType` and remove the interface completely?

Reply:
Interfaces are a feature. It will help users to inject objects using a DI. It would even help them to Mock our classes.

As far as I can see the only functionality it's implementing is `Devices`. I could even think why these aren't extension methods... just speaking my mind.

Reply:
the `ExecutablePath` is implemented.
I don't see an issue having multiple abstract methods

Reply:
fair enough.